spswarehouse/powerschool/powerschool_calpads.py

```diff
@@ -0,0 +1,311 @@
+                return self._download_calpads_eoy_report_for_school_student_absence_summary(
+                    report_name=report_name,
+                    file_postfix=file_postfix, 
+                    destination_directory_path=destination_directory_path, 
+                    report_parameters=report_parameters,
+                    validation_only_run=validation_only_run
+                    )
+            elif(report_name == "Student Program Records"):
+                return self._download_calpads_eoy_report_for_school_student_program_records(
+                    report_name=report_name,
+                    file_postfix=file_postfix, 
+                    destination_directory_path=destination_directory_path, 
+                    report_parameters=report_parameters,
+                    validation_only_run=validation_only_run
+                    )
+            elif(report_name == "Course Section Records"):
+                return self._download_calpads_eoy_report_for_school_course_section_records(
+                    report_name=report_name,
+                    file_postfix=file_postfix, 
+                    destination_directory_path=destination_directory_path, 
+                    report_parameters=report_parameters,
+                    validation_only_run=validation_only_run
+                    )
+            elif(report_name == "Student Course Section Records"):
+                return self._download_calpads_eoy_report_for_school_student_course_section_records(
+                    report_name=report_name,
+                    file_postfix=file_postfix, 
+                    destination_directory_path=destination_directory_path, 
+                    report_parameters=report_parameters,
+                    # The below is an additional parameter compared to the function calls earlier in the 
+                    #   if-else tree
+                    school_subdistrict_name=school_subdistrict_name, 
+                    validation_only_run=validation_only_run,
+                    )
+            else:
+                raise Exception("CALPADS EOY report name not supported")
+        elif(submission_window == 'All Year'):
+            if(report_name == 'SSID Enrollment Records'):
+                return self._download_calpads_all_year_report_for_ssid_enrollment_records_senr(
+                    report_name=report_name,
+                    file_postfix=file_postfix, 
+                    destination_directory_path=destination_directory_path, 
+                    report_parameters=report_parameters,
+                    validation_only_run=validation_only_run,
+                    )
+            else:
+                raise Exception("CALPADS All Year report name not supported")
+        else:
+            raise Exception("Submission window name not supported")
+        
+
+    # All Year Reports #################
+
+    def _download_calpads_all_year_report_for_ssid_enrollment_records_senr(self, file_postfix: str, 
```

Comment:
I like having the full report name here, because some of the report names are VERY close to each other (specifically, CRSC and SCSC), so it's useful to have a little more detail as a reminder of which is which. That said, I streamlined the names a bit and made sure every function has the 4-letter report code at the end. Not all of them did before. They should read a bit cleaner now.